src/editor/Core/Ripgrep.re

```diff
@@ -0,0 +1,50 @@
+let mergeOptions = options =>
+  List.fold_left((accum, opt) => accum ++ " " ++ opt, "", options);
+
+let getPath = () =>
+  Utility.(
+    Revery.Environment.(
+      join([getWorkingDirectory(), "assets", "ripgrep"])
+      |> (
+        grepDir =>
+          switch (os) {
+          | Mac => join([grepDir, "mac", "rg"])
+          | Linux => join([grepDir, "linux", "rg"])
+          | Windows => join([grepDir, "windows", "rg.exe"])
+          | Browser
+          | Unknown => failwith("not implemented")
+          }
+      )
+    )
+  );
+
+let process = args => {
+  let lines = ref([]);
+  let rgPath = getPath();
```

Comment:
Paths with spaces causing issues for me again...I guess best to find them now rather than later!

I get an error that `F:\User` is not a recoginsed command, since the full path is `F:\User Files\Documents\git\Oni2...`. Think it needs escaping, or quoting.

Reply:
hmm thats odd, the function I'm using comes from revery, I'm otherwise not really doing anything in Oni2 (here) with the actual path. Must mean the function in revery doesn't work on windows. Did the initial error occur in revery or in oni2? either way I'll find a fix but would be good to know the source

Reply:
I'll check when I'm in and see what path it is producing and how it should be formatted. Maybe `revery` should be returning an escaped path?

Out of curiosity, do you know why this way isn't used for the other vendored parts, such as nvim itself, and its instead done with the `bootstrap` script ?

Reply:
@CrossR, @bryphe initially mentioned and discussed this in https://github.com/onivim/oni2/issues/40

I think this pattern I'm using is probably going to need to change to `getExecutingDir` because I *think* if I remember correctly that once we have release build only the build output will be available so things/dependencies should be referenced from that perspective

Reply:
I've added a change to add an escape char to spaces can you let me know if that works for you @CrossR?